### PR TITLE
Added ability to use custom location for local Swift packages

### PR DESCRIPTION
**Pull request by John-Connolly**
Relates to https://github.com/yonaskolb/XcodeGen/issues/1173

### CODE REVIEW

File: Docs/ProjectSpec.md [hunk 1]
```diff
@@ -982,6 +983,9 @@ packages:
     from: 0.5.0
   RxClient:
     path: ../RxClient
+  AppFeature:
+    path: ../Packages
+    xcodePath Domains/AppFeature
```
Comment (yonaskolb) [inline Docs/ProjectSpec.md:988]: needs a `:`

File: Sources/ProjectSpec/SwiftPackage.swift [hunk 2]
```diff
@@ -90,8 +92,9 @@ extension SwiftPackage: JSONEncodable {
                 dictionary["revision"] = revision
             }
             return dictionary
-        case .local(let path):
+        case let .local(path, localPath):
             dictionary["path"] = path
+            dictionary["localPath"] = localPath
```
Comment (yonaskolb) [inline Sources/ProjectSpec/SwiftPackage.swift:97]: Using a different name

**Review by yonaskolb — COMMENTED** [review]
Thanks for this @John-Connolly, it's a great addition. Could you please:
- add a changelog entry
- rename `xcodePath` and `localPath` to `group` which is Xcode's terminology for this.
- add a test to SpecloadingTests "parses packages"
- update the fixture in Tests/Fixtures/SPM/project.yml to add a group to the XcodeGen package. (You'll need to commit the diff change after running tests)

File: Docs/ProjectSpec.md (hunk 1 above)
Comment (John-Connolly) [reply Docs/ProjectSpec.md]: Update here https://github.com/yonaskolb/XcodeGen/pull/1175/commits/a87bf32c1ed059a76377f1f09812f9b55e8a12d1

File: Sources/ProjectSpec/SwiftPackage.swift (hunk 2 above)
Comment (John-Connolly) [reply Sources/ProjectSpec/SwiftPackage.swift]: Update here https://github.com/yonaskolb/XcodeGen/pull/1175/commits/a87bf32c1ed059a76377f1f09812f9b55e8a12d1

**Comment by John-Connolly** [comment]
@yonaskolb Github really didn't want to show the correct changes without changing the base branch, but I have made the changes you mentioned :)

**Review by yonaskolb — APPROVED** [review]
Great thanks!

**Pull request merged**